src/rtstartup/rsbegin.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// rsbegin.o and rsend.o are the so called "compiler runtime startup objects".
+// They contain code needed to correctly initialize the compiler runtime.
+//
+// When an executable or dylib image is linked, all user code and libraries are
+// "sandwiched" between these two object files, so code or data from rsbegin.o
+// become first in the respective sections of the image, whereas code and data
+// from rsend.o become the last ones.  This effect can be used to place symbols
+// at the beginning or at the end of a section, as well as to insert any required
+// headers or footers.
+//
+// Note that the actual module entry point is located in the C runtime startup
+// object (usually called `crtX.o), which then invokes initialization callbacks
+// of other runtime components (registered via yet another special image section).
+
+#![feature(no_std)]
+
+#![crate_type="rlib"]
+#![no_std]
+#![allow(non_camel_case_types)]
+
+#[cfg(all(target_os="windows", target_arch = "x86", target_env="gnu"))]
+pub mod eh_frames
+{
+    #[no_mangle]
+    #[link_section = ".eh_frame"]
+    // Marks beginning of the stack frame unwind info section
+    pub static __EH_FRAME_BEGIN__: [u8; 0] = [];
```

Comment:
So just to clarify, this works because:

* The linker will construct the `.eh_frame` section from the list of items contained within it on the command line. This is, however, guarantee to be the first, so `&__EH_FRAME_BEGIN__` is guaranteed to be a pointer to the start of the section.
* Similarly, `rsend.o` is guarnateed to be at the end, so it's guaranteed that `&__EH_FRAME_END__` is at the end of the section.

Does that sound right? Also out of curiosity, what is the `__EH_FRAME_END__` symbol used for? It's not explicitly referenced here, so is it referenced or dynamically loaded from the CRT or elsewhere?